infra/src/main/kotlin/fi/fta/geoviite/infra/switchLibrary/SwitchStructure.kt

```diff
@@ -142,25 +145,28 @@ fun parseSwedishSwitchType(typeName: String): SwitchTypeParts? {
     )
 }
 
-data class SwitchType @JsonCreator(mode = DELEGATING) constructor(val typeName: String) {
-    val parts =
-        if (typeName.startsWith("EV")) {
-            parseSwedishSwitchType(typeName)
-                ?: throw IllegalArgumentException("Cannot parse switch type: \"${formatForException(typeName)}\"")
-        } else {
-            parseFinnishSwitchType(typeName)
+data class SwitchType
+@JsonCreator(mode = DELEGATING)
+private constructor(val typeName: String, val parts: SwitchTypeParts) {
```

Comment:
Toinen rakentaja -> .of muutos on vaihdetyypille. Tietty string-muotoinen vaihdetyyppinimi on toki aina sama vaihdetyyppi ja noita on hyvin rajallinen määrä -> ikuinen cache. Tässäkin olion luonti ei ole se pointti vaan tuo tyypin osien parsinta. Se näkyi flame graafissa (jälleen pieni puro, joista kertyy).